analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/CallModelStateIsValid.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                compilationStart.RegisterSymbolStartAction(symbolStart =>
+                {
+                    var type = (INamedTypeSymbol)symbolStart.Symbol;
+                    if (type.DerivesFrom(KnownType.Microsoft_AspNetCore_Mvc_ControllerBase)
+                        && !type.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ApiControllerAttribute)
+                        && !HasActionFilterAttribute(type))
+                    {
+                        symbolStart.RegisterSyntaxNodeAction(ProcessControllerMethods, SyntaxKind.MethodDeclaration);
+                    }
+                }, SymbolKind.NamedType);
+            }
+        });
+
+    private static void ProcessControllerMethods(SonarSyntaxNodeReportingContext context)
+    {
+        var method = (MethodDeclarationSyntax)context.Node;
+        if (context.SemanticModel.GetDeclaredSymbol(context.Node) is IMethodSymbol methodSymbol
+            && methodSymbol.IsControllerMethod()
+            && !HasActionFilterAttribute(methodSymbol)
+            && methodSymbol.Parameters.Any(x => HasValidationAttribute(x) || IsValidatable(x.Type)))
+        {
+            var walker = new ModelStateFinder(context.SemanticModel);
+            walker.Visit(method.Body);
+            if (!walker.ValidatesModel)
+            {
+                context.ReportIssue(Rule, method.Identifier);
+            }
+        }
+    }
+
+    private static bool IsValidatable(ITypeSymbol type) => type != null
+        && (type.Implements(KnownType.System_ComponentModel_DataAnnotations_IValidatableObject)
+            || type.GetMembers().OfType<IPropertySymbol>().Any(HasValidationAttribute)
+            || IsValidatable(type.BaseType));
+
+    private static bool HasValidationAttribute(ISymbol symbol) =>
+        symbol.GetAttributes().Any(x => x.AttributeClass.DerivesFrom(KnownType.System_ComponentModel_DataAnnotations_ValidationAttribute));
+
+    private static bool HasActionFilterAttribute(ISymbol symbol) =>
+        symbol.GetAttributes().Any(x => x.AttributeClass.DerivesFrom(KnownType.Microsoft_AspNetCore_Mvc_Filters_ActionFilterAttribute));
+
+    private sealed class ModelStateFinder : SafeCSharpSyntaxWalker
+    {
+        private readonly SemanticModel semanticModel;
+        public bool ValidatesModel { get; private set; }
+
+        public ModelStateFinder(SemanticModel semanticModel) => this.semanticModel = semanticModel;
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!ValidatesModel)
+            {
+                base.Visit(node);
+            }
+        }
+
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+        {
+            if (node.GetName() == "TryValidateModel"
+               && semanticModel.GetSymbolInfo(node.Expression).Symbol is IMethodSymbol method
+               && method.ContainingType.Is(KnownType.Microsoft_AspNetCore_Mvc_ControllerBase))
```

Comment:
[TryValidateModel](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.tryvalidatemodel?view=aspnetcore-8.0) is virtual. Add a test case where `TryValidateModel` is overridden.